### ISSUE TITLE
Proper Benchmark designing on several survival analysis tasks

### ISSUE THREAD

**Issue by steletvinicius** [mlr-org/mlr3#1396]
Hello, and congratulations on mlr3verse's elegant and efficient interface.
My team and I decided to invest some time on learning its particular notation as it seems to be promissing and really useful.

I'm pursuing my PhD and have been conducting experiments with machine learning and survival analysis on a very large dataset.

I've already familiarized myself with some of the mlr3verse notation and would like to simplify things by using the benchmark function to run all the tests on both the overall dataset and its segments.

I want to test 13 possible ML algorithms using Cox ph as my performance comparison algorithm.

For each task and model, I would like to:

A) on the training set:
1. Find the best hyperparameter configuration using cross-validation, using the Concordance index as a performance metric;
2. Train the final model;

B) On the test set:

Measure the performance of all models in not only one but 30 resamples with replacement (bootstraping), using the C-index and other survival metrics as a comparative metrics.

I would appreciate your help on how to write this experiment using the proper benchmark function notation.

Here is part of my script showing what I have done so far:

[https://gist.github.com/steletvinicius/300cfc81a1e85e147335be0f5ad4da0c](https://github.com/mlr-org/mlr3benchmark/issues/url)

Some questions:

Does benchmark function uses only the train partition to run the hyperparameter optimization?
How to access for each task the best fitted model performance on the training and the test sets?
How to extract the best models fromt he benchmark for future steps, like feature importance evaluations and to release on the production environment?

**Comment by bblodfon** [comment]
Hi @steletvinicius, 

First, thanks for trying `mlr3` out, and specifically the survival (`mlr3proba` currently) part of it. I would suggest, if the benchmark is really large-scale, or you want some starting code for inspiration, that you check the repo that @jemus42 has made for a publication here => https://github.com/slds-lmu/paper_2023_survival_benchmark. The chapter of the `mlr3` book => https://mlr3book.mlr-org.com/chapters/chapter11/large-scale_benchmarking.html might come useful as well.

Some quick answers to your questions (please post `reprex()` small code snippets for more specific questions so that we can easily check and answer these):
> Does benchmark function uses only the train partition to run the hyperparameter optimization?

Yes, for each train set as defined by the resampling strategy of your choice (and for a particular task/learner combo), the learner (`AutoTuner`) will be tuned and have a final model, with best hyperparameters, for testing in the corresponding test set.

> How to access for each task the best fitted model performance on the training and the test sets?

Please read => https://mlr3book.mlr-org.com/chapters/chapter15/predsets_valid_inttune.html#sec-predict-sets

> How to extract the best models fromt he benchmark for future steps, like feature importance evaluations and to release on the production environment?

There might be other ways, but for a particular resampling $i$, you can do: 
```
# `bmr` => result of `benchmark(...)`
trained_learners = bmr$resample_result(i)$learners
lapply(trained_learners, function(learner) { learner$importance() })
```

**Comment by jemus42** [comment]
Note that the linked benchmark repo for our publication has quite a few "extra bits" but most likely exceed what's needed to get started 😅 

For example:

- We use `mlr3batchmark` as glue between mlr3 and [batchtools](https://github.com/mlr-org/batchtools) to run things on an HPC, which is a tad more complex than "just" running `benchmark()`
- We add tuning callbacks to keep logs and manually write the tuning archives to disk, which is just a workaround to avoid storing them in the `BenchmarkResult` object, which would unfortunately drastically increase the size of the object in RAM
- We tune every learner twice, once with C-index and once with the integrated survival Brier score, you probably only want to tune everything once 😬 
- The entire setup for the experiment had evolved and become more complex over many years, sorry about the mess 😅

**Comment by bblodfon** [comment]
Hi @steletvinicius , have our answers be helpful? This is more a stackoverflow type of question and not an issue, so I will soon close this issue.

**Comment by steletvinicius** [comment]
Yes, truly helpful. Thanks @bblodfon
